app/lib/llm_clients/openai/chat_request_helper.rb

```diff
@@ -0,0 +1,50 @@
+require "openai"
+
+module LlmClients
+  module Openai
+    module ChatRequestHelper
+      def chat_request(chat_history, &)
+        @stats = new_stats
+        stats[:start_time] = current_time
+        num_tokens = 0
+
+        current_batch = ""
+        current_batch_size = 0
+
+        resp = chat_client.chat(
+          parameters: {
+            model: @model,
+            messages: chat_history, # Required.
+            temperature: @temperature,
+            stream: proc do |chunk|
+                      if num_tokens.zero?
+                        stats[:first_token_time] = current_time
+                      end
+                      num_tokens += 1
+                      unless (str = chunk.dig("choices", 0, "delta", "content")).nil?
+                        current_batch << str
+                        current_batch_size += 1
+                        if str.ends_with?("\n") || str.ends_with?("}") || current_batch_size >= @batch_size
+                          Rails.logger.debug { "==> #{current_batch}" }
+                          yield current_batch
+
+                          current_batch_size = 0
```

Comment:
Ruby's `String#length` method is BigO(1) (unless the string is multi-byte encoded), so you could dispense with this counter and just use `current_batch.length` if you want.

Reply:
I based this on how the batch size was implemented in the `ollama` client, where it looks like the batch size is "token count" vs number of characters. See lines 55-59 in `llm_clients/ollama/client.rb`.

So decision here could go two ways:
1. It should have been character count, in which case I'll switch to use `#size`
2. It should be token count, in which case I can refactor to rename things to better reflect intent.

Thoughts?

Reply:
It's fine.